### PR TITLE
Normalize domains to fqdn for resolver selection

### PR DESCRIPTION
**Pull request by vyzo**

### CODE REVIEW

**Review by lidel — APPROVED** [review]
left-to-right checks coupled with the use of `dns.Fqdn` lgtm

File: resolve.go [hunk 1]
```diff
@@ -63,23 +64,31 @@ func WithDomainResolver(domain string, rslv BasicResolver) Option {
...
 	}
 
-	for i := strings.Index(domain, "."); i != -1; i = strings.Index(domain, ".") {
-		domain = domain[i+1:]
-		rslv, ok = r.custom[domain]
+	for i := strings.Index(fqdn, "."); i != -1; i = strings.Index(fqdn, ".") {
+		fqdn = fqdn[i+1:]
+		if fqdn == "" {
+			// the . is the default resolver
+			break
```
Comment (aschmahmann) [inline resolve.go:88]: Is there a test for this?

**Review by aschmahmann — APPROVED** [review]
LGTM

File: resolve.go (hunk 1 above)
Comment (vyzo) [reply resolve.go:88]: not explicitly, but it should get exercised in every miss.

**Pull request merged**